### PR TITLE
console: implement cutoff for large arrays

### PR DESCRIPTION
**Pull request by nektro**
added new test coverage as well

### CODE REVIEW

**Comment by robobun** [comment]
@nektro, your commit 27337245c7a0c0c25da3ce1ed519f95a991d8290 is building: [`#1680`](https://buildkite.com/bun/bun/builds/1680)

File: src/bun.js/ConsoleObject.zig [hunk 1]
```diff
@@ -2195,6 +2197,13 @@ pub const Formatter = struct {
                             }
                             continue;
                         }
+                        if (j >= 100) {
+                            writer.writeAll(",\n");
```
Comment (Jarred-Sumner) [inline src/bun.js/ConsoleObject.zig]: this should be goodTimeForANewLine() and use printComma

Comment (Jarred-Sumner) [reply src/bun.js/ConsoleObject.zig]: i think

File: src/bun.js/ConsoleObject.zig [hunk 2]
```diff
@@ -2186,6 +2187,7 @@ pub const Formatter = struct {
                     }
 
                     var i: u32 = 1;
+                    var j: u32 = 1;
```
Comment (Jarred-Sumner) [inline src/bun.js/ConsoleObject.zig]:
```suggestion
                    var non_holey_array_items_count: u32 = 1;
```

**Pull request merged**